src/tls/openssl.c

```diff
@@ -724,62 +752,64 @@ static int tls_net_handshake(struct flb_tls *tls,
         }
     }
 
+    // Clear any previous SSL errors
     ERR_clear_error();
 
+    // Perform SSL handshake
     if (tls->mode == FLB_TLS_CLIENT_MODE) {
         ret = SSL_connect(session->ssl);
     }
     else if (tls->mode == FLB_TLS_SERVER_MODE) {
         ret = SSL_accept(session->ssl);
     }
 
-    if (ret != 1) {
-        ret = SSL_get_error(session->ssl, ret);
-        if (ret != SSL_ERROR_WANT_READ &&
-            ret != SSL_ERROR_WANT_WRITE) {
-            ret = SSL_get_error(session->ssl, ret);
-            // The SSL_ERROR_SYSCALL with errno value of 0 indicates unexpected
-            //  EOF from the peer. This is fixed in OpenSSL 3.0.
-            if (ret == 0) {
+    // Exit early if the handshake was successful
+    if (ret == 1) {
```

Comment:
:+1:
I just confused the condition but it is correct. `SSL_accept` returns 1 when connection is established.
https://docs.openssl.org/master/man3/SSL_accept/#return-values